sqe-database-access/SignInterpretationRepository.cs

```diff
@@ -39,16 +34,19 @@ public async Task<SignInterpretationData> GetSignInterpretationById(UserInfo use
         {
             // We use several existing quick functions to get the specifics of a sign interpretation,
             // so wrap it in a transaction to make sure the result is consistent.
-            using (var transactionScope = new TransactionScope())
+            using (var transactionScope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))
             using (var conn = OpenConnection())
             {
                 var attributes =
                     _attributeRepository.GetSignInterpretationAttributesByInterpretationId(user, signInterpretationId);
                 var commentaries =
                     _interpretationCommentaryRepository.GetSignInterpretationCommentariesByInterpretationId(user,
                         signInterpretationId);
-                var roiIds = await _roiRepository.GetSignInterpretationRoisIdsByInterpretationId(user, signInterpretationId);
-                var rois = roiIds.Select(x => _roiRepository.GetSignInterpretationRoiByIdAsync(user, x));
+                var roiIds =
+                    await _roiRepository.GetSignInterpretationRoisIdsByInterpretationId(user, signInterpretationId);
+                var rois = new SignInterpretationRoiData[roiIds.Count];
+                foreach (var (roiId, index) in roiIds.Select((x, idx) => (x, idx)))
```

Comment:
Such loops, in case they pose a performance issue, are prime candidates for a different query - one that returns all ROIs in one query.

Reply:
Yeah, we should probably add a method that does such a query.  In this case I figured we would usually have only one or two ROI's, so I was lazy and reused existing functionality.  I am putting this as a TODO comment in the next PR.